packages/amplify-util-mock/src/func/index.ts

```diff
@@ -58,7 +58,7 @@ export async function start(context: $TSContext): Promise<void> {
       printer.info(typeof result === 'undefined' ? '' : stringResult);
     } catch (err) {
       printer.error(`${resourceName} failed with the following error:`);
-      printer.info(err);
+      printer.info(err?.message);
```

Comment:
instead of doing this. could we modify functions in printer and make them accept `string | Error` ?
Then depending on input type extract apply desired strategy?

See https://github.com/aws-amplify/amplify-cli/blob/fd51b652c3fb6b24db7a400bb38bf5a35757efc2/packages/amplify-prompts/src/printer.ts#L11-L35 .

additionally. 
why do we `printer.error` first message and `printer.info` second line? Should this all be `printer.error` ?

Reply:
btw. if the answer to second concern is `yes` then we should only consider modifying `printer.error` to accept both string and error.

Reply:
@lazpavel @jhockett what do you think ?

Reply:
Then perhaps we need
`error = (line: string, error?: Error): void` on printer , i.e. add error as optional parameter and make sure it prints nicely together.

Reply:
I like both suggestion, was thinking about info/error myself but didn't comment on it since it was an existing behavior. But in general we should keep errors to error console as someone can split the outputs

Reply:
updated the error according to the suggestions

Reply:
I missed this thread until now, but I talked with @ykethan and I think the suggestion makes sense